app/models/carto/layer.rb

```diff
@@ -229,6 +229,8 @@ def default_query(user = nil)
 
         if table_name.present? && !table_name.include?('.') && user_name.present? && qualify
           "SELECT * FROM #{safe_table_name_quoting(user_name)}.#{safe_table_name_quoting(table_name)}"
+        elsif database_schema.present?
+          "SELECT * FROM #{safe_table_name_quoting(database_schema)}.#{safe_table_name_quoting(table_name)}"
         else
           "SELECT * FROM #{qualified_table_name}"
```

Comment:
The `qualified_table_name` method already adds the schema prefix when passing the owner, so I think this should be equivalent (you could also remove the new parameter):
```suggestion
        else
          "SELECT * FROM #{qualified_table_name(user)}"
```

Also, we could refactor this method to avoid checking if it's an organization user and always use the database schema from the user... But this is quite tricky and maybe it's too much for now :sweat_smile: